Author: <mstone@debian.org>
Description: timeout ignores fractional part of sleep times when timeout is more 
than 100000s (approximately 1 day) on kfbsd. prevents failure modes 
in libc implementation when timeout approaches max(time_t)
Index: coreutils/src/timeout.c
===================================================================
--- coreutils.orig/src/timeout.c	2020-07-24 13:38:46.287061953 -0400
+++ coreutils/src/timeout.c	2020-07-24 13:38:46.283061984 -0400
@@ -122,6 +122,11 @@
    resolution provided by alarm().  */
 
 #if HAVE_TIMER_SETTIME
+#ifdef __FreeBSD_kernel__
+if (duration < 100000) {
+#else
+if (true) {
+#endif 
   struct timespec ts = dtotimespec (duration);
   struct itimerspec its = { {0, 0}, ts };
   timer_t timerid;
@@ -138,6 +143,7 @@
     }
   else if (warn && errno != ENOSYS)
     error (0, errno, _("warning: timer_create"));
+}
 #endif
 
   unsigned int timeint;
